main.go

```diff
@@ -85,7 +88,21 @@ func main() {
 	log.Entry(ctx).Infof("Config: %#v", config)
 
 	// Get a X509Source
-	source, err := workloadapi.NewX509Source(ctx)
+	source, err := workloadapi.NewX509Source(ctx,
+		workloadapi.WithClientOptions(workloadapi.WithLogger(log.Entry(ctx))),
+		workloadapi.WithDefaultX509SVIDPicker(func(svids []*x509svid.SVID) *x509svid.SVID {
```

Comment:
So I'm not clear why we are using the SVID Picker here...

Reply:
Actually the deployment can fetch few svids. By default `workloadapi.NewX509Source` just picks first.
So I think we should pick svid for registry if for that has been deployed service account.

Reply:
Hmm... I'm inclined to not worry about it right now... in practice at least in K8s, I believe its only returning *one* SVID, and this *forces* an SVID schema on users they may not want.

Reply:
> in practice at least in K8s, I believe its only returning one SVID

I guess it is related to user spire setup...